tests/data_test.py

```diff
@@ -30,15 +32,48 @@ def test_datasets(ds_cls: type[VisionDataset]) -> None:
 
 
 def test_audio_dataset() -> None:
-    """Tests basic functionality of an audio dataset base class."""
+    """Tests basic functionality of the base audio dataset class."""
     x = torch.rand(1, 10)
     audio_dir = Path(r"Sample path")
     dataset = PBAudioDataset(x=x, audio_dir=audio_dir)
 
     assert dataset is not None
     assert len(dataset) == len(x)
+
+    
+def test_audio_dataset() -> None:
+    root_dir = Path("~/Data").expanduser()
+    metadata = pd.read_csv(root_dir / "EcoacousticsDS" / "metadata.csv")
+    target_attribute = "habitat"
+
+    ds_cls_dnwld = EcoacousticsDS(root=root_dir, target_attr=target_attribute)
+    assert ds_cls_dnwld is not None
+
+    ds_cls_no_dnwld = EcoacousticsDS(root=root_dir, download=False, target_attr=target_attribute)
+    assert ds_cls_no_dnwld is not None
+
+    # Test __len__
+    num_audio_samples = []
+    num_audio_samples.extend(root_dir.glob("**/*.wav"))
```

Comment:
```suggestion
    num_audio_samples = list(root_dir.glob("**/*.wav"))
```

Reply:
Much nicer, I'm relying too much on the copy paste but was also lazy not checking the return of `glob` thought it might have necessitated extend by being a list or something.